libcloud/compute/drivers/vsphere.py

```diff
@@ -49,506 +36,228 @@
     def list_nodes(self):
-        vm_paths = self.connection.client.get_registered_vms()
-        nodes = self._to_nodes(vm_paths=vm_paths)
-
-        return nodes
-
-    @wrap_non_libcloud_exceptions
-    @wrap_non_libcloud_exceptions
-    def ex_clone_node(self, node, name, power_on=True, template=False):
-        """
-        Clone the provided node.
-
-        :param node: Node to clone.
-        :type node: :class:`libcloud.compute.base.Node`
-
-        :param name: Name of the new node.
-        :type name: ``str``
-
-        :param power_on: Power the new node on after being created.
-        :type power_on: ``bool``
-
-        :param template: Specifies whether or not the new virtual machine
-                         should be marked as a template.
-        :type template: ``bool``
-
-        :return: New node.
-        :rtype: :class:`libcloud.compute.base.Node`
         """
-        vm = self._get_vm_for_node(node=node)
-        new_vm = vm.clone(name=name, power_on=power_on, template=template)
-        new_node = self._to_node(vm=new_vm)
-
-        return new_node
-
-    @wrap_non_libcloud_exceptions
-    def ex_migrate_node(self, node, resource_pool=None, host=None,
-                        priority='default'):
+        Lists nodes
         """
-        Migrate provided node to a new host or resource pool.
+        nodes = []
+        content = self.connection.RetrieveContent()
+        children = content.rootFolder.childEntity
+        for child in children:
+            if hasattr(child, 'vmFolder'):
+                datacenter = child
+            else:
+            # some other non-datacenter type object
+                continue
+            vm_folder = datacenter.vmFolder
+            vm_list = vm_folder.childEntity
+
+            for virtual_machine in vm_list:
+                node = self._to_node(virtual_machine, 10)
+                if node:
+                    nodes.append(node)
+        return nodes
 
-        :param node: Node to clone.
-        :type node: :class:`libcloud.compute.base.Node`
+    def _to_node(self, virtual_machine, depth=1):
+        maxdepth = 10
+        # if this is a group it will have children.
+        # if it does, recurse into them and then return
+        if hasattr(virtual_machine, 'childEntity'):
+            if depth > maxdepth:
+                return
+            vmList = virtual_machine.childEntity
+            for c in vmList:
+                self._to_node(c, depth + 1)
+            return
 
-        :param resource_pool: ID of the target resource pool to migrate the
-                              node into.
-        :type resource_pool: ``str``
+        summary = virtual_machine.summary
+        name = summary.config.name
+        path = summary.config.vmPathName
+        memory = summary.config.memorySizeMB
+        cpus = summary.config.numCpu
+        operating_system = summary.config.guestFullName
+        # mist.io needs this metadata
+        os_type = 'unix'
```

Comment:
can you make that a ternary statement please. also remove the comment about mist